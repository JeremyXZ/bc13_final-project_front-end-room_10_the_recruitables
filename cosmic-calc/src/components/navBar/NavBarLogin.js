import React from "react";
import { Nav, NavLink, NavMenu } from "./navBarElements";
import logo from "./logo.png";
import "./logo.css";
// import Logout from "../Authentication/Logout";
//if use is not logged in display login and register buttons and if they are logged in display the profile

const NavBarLogin = () => {
  return (
    <>
      <Nav>
        <img src={logo} alt="logo" className="logo" />
        <NavMenu>
          <NavLink to="/" activeStyle>
            login
          </NavLink>
          <NavLink to="/register" activeStyle>
            register
          </NavLink>
        </NavMenu>
      </Nav>
    </>
  );
};

export default NavBarLogin;
