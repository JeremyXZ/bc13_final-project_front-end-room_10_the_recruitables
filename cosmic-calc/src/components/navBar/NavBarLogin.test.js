import React from "react";
import "@testing-library/jest-dom";
import {screen, render } from "@testing-library/react";
import {BrowserRouter, Router} from 'react-router-dom';
import userEvent from "@testing-library/user-event";

import NavBarLogin from "./NavBarLogin";

describe("<NavBarLogin/>", () => {
    it("successfully renders the component, matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <NavBarLogin />
            </BrowserRouter>
        );
        expect(asFragment()).toMatchSnapshot();
      });

    it('renders all 2 links on the screen', () => {
        render(
            <BrowserRouter>
                <NavBarLogin />
            </BrowserRouter>
        );
       
        const links = screen.getAllByRole('link')
        expect(links.length).toBe(2)
    })


    it('renders all 2 link texts on the screen', () => {
        const {getAllByText} = render(
            <BrowserRouter>
                <NavBarLogin />
            </BrowserRouter>
        );
        expect(getAllByText(/login/i)[0]).toBeInTheDocument();
        expect(getAllByText(/register/i)[0]).toBeInTheDocument();
    })
   
    
})