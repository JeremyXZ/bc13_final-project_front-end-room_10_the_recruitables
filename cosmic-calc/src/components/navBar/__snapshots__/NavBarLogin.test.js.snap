// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBarLogin/> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="sc-bcXHqe llTDYN"
  >
    <img
      alt="logo"
      class="logo"
      src="logo.png"
    />
    <div
      class="sc-dkrFOg fLDoxR"
    >
      <a
        aria-current="page"
        class="sc-gswNZR kjkvCA active"
        href="/"
      >
        login
      </a>
      <a
        class="sc-gswNZR kjkvCA"
        href="/register"
      >
        register
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
